packages/core/database/lib/entity-manager/index.js

```diff
@@ -827,18 +832,47 @@ const createEntityManager = (db) => {
 
               // add order value
               if (hasOrderColumn(attribute)) {
-                const orderMax = (
-                  await this.createQueryBuilder(joinTable.name)
-                    .max(orderColumnName)
-                    .where({ [joinColumn.name]: id })
-                    .where(joinTable.on || {})
-                    .first()
-                    .transacting(trx)
-                    .execute()
-                ).max;
+                // TODO : Merge with the one below
+                const adjacentRelations = await this.createQueryBuilder(joinTable.name)
+                  .where({
+                    [joinColumn.name]: id,
+                    [inverseJoinColumn.name]: {
+                      $in: compact(
+                        cleanRelationData.connect.map(
+                          (r) => r.position?.after || r.position?.before
+                        )
+                      ),
+                    },
+                  })
+                  .transacting(trx)
+                  .execute();
 
-                insert.forEach((row, idx) => {
-                  row[orderColumnName] = orderMax + idx + 1;
+                const maxOrder = await this.createQueryBuilder(joinTable.name)
+                  .max(orderColumnName)
+                  .where({ [joinColumn.name]: id })
+                  .where(joinTable.on || {})
+                  .first()
+                  .transacting(trx)
+                  .execute();
```

Comment:
@petersg83  Do you think we could optimize this ? I do not like having two queries here.